gcovr/writer/html.py

```diff
@@ -405,6 +404,14 @@ def print_html_report(covdata, output_file, options):
 
         data['filename'] = cdata_fname[f]
 
+        # Only use demangled names (containing a brace)
+        data['function_names'] = sorted(
+            [
+                (f, cdata.functions[f].lineno)
+                for f in cdata.functions.keys() if ("(" in f) or (f == "main")
```

Comment:
I don't think demangled names have to contain a paren `()`. In particular, plain C function names will not contain one (since they aren't mangled to start with). So I would remove this filtering.

Reply:
Ok, now I see that in the test there is nearly all C++. I'll take a look into this. I added the filter because in clang we don't have the option `--demangle-names` and a name like e.g. `__Zx6foo5` isn't usable.

Reply:
I'd rather show a mangled name than accidentally hide valuable information. Separately from this PR, we can figure out a way to ensure as much demangling as possible. For example, using `_cxa_demangle()` function from libc++ if available, or piping through `c++filt` from GNU binutils or llvm-cxxfilt.